### PR TITLE
 feat(auth): Implementation of hostedUI flow

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:*

*Description of changes:*

Added implementation for the hosted UI flows.

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                } else if let idToken = json["id_token"] as? String,
+                          let accessToken = json["access_token"] as? String,
+                          let refreshToken = json["refresh_token"] as? String,
+                          let expiresIn = json["expires_in"] as? Int {
+                    let userPoolTokens = AWSCognitoUserPoolTokens(idToken: idToken,
+                                                                  accessToken: accessToken,
+                                                                  refreshToken: refreshToken,
+                                                                  expiresIn: expiresIn)
+                    let signedInData = SignedInData(userId: "",
+                                                    userName: "",
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift]: Fixed

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                                                    signedInDate: Date(),
+                                                    signInMethod: .apiBased(.userSRP),
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift]: Fixed

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        }
+    }
+}
+
+
+
+extension FetchHostedUISignInToken: CustomDebugDictionaryConvertible {
+    var debugDictionary: [String: Any] {
+        [
+            "identifier": identifier
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift]: Added

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            else if let data = data {
+                handleData(data, dispatcher: dispatcher, environment: environment)
+            } else {
+                let signInError = SignInError.unknown(message: "Could not fetch Token: \(urlResponse.debugDescription)")
+                let event = HostedUIEvent(eventType: .throwError(signInError))
+                logVerbose("\(#fileID) Sending event \(event.type)", environment: environment)
+                dispatcher.send(event)
+            }
+        }
+        task.resume()
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/HostedUI/FetchHostedUISignInToken.swift:87]: Will change when we make the public api async, else there is no point in using async and then converting it back to callback.

**Review by harsh62 — APPROVED** [review]

**Pull request merged**